odl/trafos/non_uniform_fourier.py

```diff
@@ -9,123 +9,242 @@
         """Initialize a new instance.
 
         Parameters
         ----------
-        space : DiscreteLp
-            The uniform space in which the data lies
+        domain : DiscreteLp
+            Uniformly discretized space where the Fourier transform takes
+            its inputs.
         samples : array-like
-            List of the fourier space positions where the coefficients are
-            computed.
+            Array of points at which the non-uniform FFT should be evaluated.
+            It must have shape ``(M, d)``, where ``M`` is the number of
+            samples and ``d`` the dimension of ``domain``. The magnitude
+            of the samples may not exceed ``pi / s``, where ``s`` is
+            ``domain.cell_sides``.
+        range : `TensorSpace`, optional
+            Range of the non-uniform FFT operator. It must be a
+            one-dimensional space with shape ``(M,)``. By default, it is
+            chosen as ``cn(M)`` with the same arithmetic precision as
+            ``domain.dtype``.
         skip_normalization : bool, optional
-            Whether the normalization step should be skipped
+            Whether the samples normalization step should be skipped.
+            If ``True``, all ``samples`` are expected to lie between -0.5
+            (inclusive) and +0.5 (exclusive).
+        kwargs
+            Additional keyword arguments passed on to the ``NFFT`` class
+            constructor, except for its ``N`` and ``M`` parameters.
+
+        Notes
+        -----
+        - The ``pynfft`` backend library currently only supports double
+          precision arithmetic, thus all computations are carried out with
+          double precision irrespective of the ``dtype`` of ``domain`` and
+          ``range``.
         """
-        if not isinstance(space, DiscreteLp) or not space.is_uniform:
-            raise ValueError("`space` should be a uniform `DiscreteLp`")
         super(NonUniformFourierTransform, self).__init__(
-            space=space,
-            samples=samples,
-            domain=space,
-            range=cn(len(samples)),
-            skip_normalization=skip_normalization,
-        )
-
-    @property
-    def adjoint(self):
-        return NonUniformFourierTransformAdjoint(
-            shape=self.shape,
-            samples=self.samples,
-            skip_normalization=True,
+            domain, samples, range, **kwargs
         )
 
     def _call(self, x):
-        """Compute the direct non uniform FFT.
-
-        Parameters
-        ----------
-        x : `numpy.ndarray`
-            The data whose non uniform FFT you want to compute
-
-        Returns
-        -------
-        out_normalized : `numpy.ndarray`
-            Result of the transform
-        """
+        """Return ``self(x)``."""
         if not self._has_run:
-            self._normalize()
+            if not self._skip_normalization:
+                normalize_samples(
```

Comment:
I just noticed that you are calling the function but not assigning the result. I think that `normalize_samples` is not inplace, so it should be `self.samples = normalize_samples(...`

Reply:
That should be taken care of by using `out=self.samples`. But it seems like a good idea anyway whether it actually does the normalization.

Reply:
I don't think so because `out` is never re-assigned in `normalize_samples`

Reply:
It is, [here](https://github.com/zaccharieramzi/odl/pull/1/files#diff-41e58e2ad3e0e48b7f781172b323a931R101).

Reply:
Right, I am really not used to this way of doing things that's why I missed it. Sorry for the unnecessary annoyance.

Reply:
Never mind, it's anyway good to have several eyes on the code, easy to make mistakes here.